recursion_exercise/recursion.js

```diff
@@ -0,0 +1,56 @@
+function productOfArray(arr) {
+    if(arr.length === 1) return arr[0]
+    return arr[0] * productOfArray(arr.slice(1))
+}
+
+
+function collectStrings(obj) {
+    var newArr = [];
+
+    function collectStringsHelper(obj) {
+        for(var i=0; i<Object.keys(obj).length; i++) {
```

Comment:
I would only call Object.keys(obj) once.  Save it to a variable and put it outside of the for loop.  Your code would then be a little more readable:

```
var keys = Object.keys(obj);
for (var i = 0; i < keys.length; i++) {
    if (typeof obj[keys[i]] === 'string') {
         newArr.push(obj[keys[i]]);
    else {
        return collectStringsHelper(obj[keys[i]]);
    }
}
```